### PR TITLE
Add Table of Contents in Docs overview

### PR DESCRIPTION
**Pull request by willium**
Add script to generate a global docs TOC file in markdown

added `npm run build:toc`

### CODE REVIEW

**Comment by willium** [comment]
screen shot 2016-04-12 at 3 38 02 am

**Comment by willium** [comment]
This is a little quick and dirty, the code's not beautiful, but hey it works.

**Comment by domoritz** [comment]
Sweet. Could you change one thing and put the toc into a markdown file in `_includes` so that we can include it anywhere with `{% include doc_toc.md %}`? I think I prefer to have to just below the overview on the first page rather than on a separate page.

**Comment by willium** [comment]
Sure. I thought that'd make it harder to notice the other sections on overview (they'd be pushed down the page)

**Comment by domoritz** [comment]
Thanks! We can still move the include but it'll be much easier if the doc toc is an include.

**Comment by kanitw** [comment]
Screenshot looks awesome!  Great work!

File: scripts/generate-toc [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env node
+'use strict';
+
+var fs = require('fs'),
+		cheerio = require('cheerio'),
+		yamlFront = require('yaml-front-matter');
+
+var URL = 'https://vega.github.io/vega-lite/docs/';
```
Comment (domoritz) [inline scripts/generate-toc]: Could you change this to `{{site.baseurl}}/docs/`?

**Comment by willium** [comment]
Fixed @domoritz!

**Pull request merged**